ckks/bootstrap_params.go

```diff
@@ -33,229 +17,1085 @@ const (
+			Qi: []uint64{
+				0x100000000060001, // 56 StC (28 + 28)
+				0xffa0001,         // 28 StC
+			},
+			ScalingFactor: [][]float64{
+				{268435456.0007324, 268435456.0007324},
+				{0xffa0001},
+			},
+		},
+		SineEvalModuli: SineEvalModuli{
+			Qi: []uint64{
+				0xffffffffffc0001,  // 60 Sine (double angle)
+				0x10000000006e0001, // 60 Sine (double angle)
+				0xfffffffff840001,  // 60 Sine (double angle)
+				0x1000000000860001, // 60 Sine (double angle)
+				0xfffffffff6a0001,  // 60 Sine
+				0x1000000000980001, // 60 Sine
+				0xfffffffff5a0001,  // 60 Sine
+				0x1000000000b00001, // 60 Sine
+				0x1000000000ce0001, // 60 Sine
+				0xfffffffff2a0001,  // 60 Sine
+				0xfffffffff240001,  // 60 Sine
+				0x1000000000f00001, // 60 Sine
+			},
+			ScalingFactor: 1 << 60,
+		},
+		CoeffsToSlotsModuli: CoeffsToSlotsModuli{
+			Qi: []uint64{
+				0x200000000e0001, // 53 CtS
+				0x20000000140001, // 53 CtS
+				0x20000000280001, // 53 CtS
+				0x1fffffffd80001, // 53 CtS
+			},
+			ScalingFactor: [][]float64{
+				{0x200000000e0001},
+				{0x20000000140001},
+				{0x20000000280001},
+				{0x1fffffffd80001},
+			},
+		},
 		H:            32768,
 		SinType:      Cos2,
-		SinRange:     257,
-		SinDeg:       250,
-		SinRescal:    3,
-		CtSLevel:     []uint64{26, 25, 24, 23},
-		StCLevel:     []uint64{11, 10, 10},
+		MessageRatio: 256.0,
+		SinRange:     325,
+		SinDeg:       255,
+		SinRescal:    4,
+		ArcSineDeg:   0,
 		MaxN1N2Ratio: 16.0,
+		BitReversed:  false,
 	},
 
-	// Set IV
-	// 768 - 110
+	// Set V
+	// 768
 	{
+		LogN:     15,
+		LogSlots: 14,
+		Scale:    1 << 25,
+		Sigma:    DefaultSigma,
+		ResidualModuli: []uint64{
+			0x1fff90001,     // 32 Q0
+			0x4000000420001, // 50
+			0x1fc0001,       // 25
+		},
+		KeySwitchModuli: []uint64{
+			0x7fffffffe0001, // 51
+			0x8000000110001, // 51
+		},
+		SlotsToCoeffsModuli: SlotsToCoeffsModuli{
+			Qi: []uint64{
+				0xffffffffffc0001, // 60 StC (30+30)
+			},
+			ScalingFactor: [][]float64{
+				{1073741823.9998779, 1073741823.9998779},
+			},
+		},
+		SineEvalModuli: SineEvalModuli{
+			Qi: []uint64{
+				0x4000000120001, // 50 Sine
+				0x40000001b0001, // 50 Sine
+				0x3ffffffdf0001, // 50 Sine
+				0x4000000270001, // 50 Sine
+				0x3ffffffd20001, // 50 Sine
+				0x3ffffffcd0001, // 50 Sine
+				0x4000000350001, // 50 Sine
+				0x3ffffffc70001, // 50 Sine
+			},
+			ScalingFactor: 1 << 50,
+		},
+		CoeffsToSlotsModuli: CoeffsToSlotsModuli{
+			Qi: []uint64{
+				0x1fffffff50001, // 49 CtS
+				0x1ffffffea0001, // 49 CtS
+			},
+			ScalingFactor: [][]float64{
+				{0x1fffffff50001},
+				{0x1ffffffea0001},
+			},
+		},
 		H:            192,
 		SinType:      Cos1,
-		SinRange:     21,
-		SinDeg:       52,
+		MessageRatio: 256.0,
+		SinRange:     25,
+		SinDeg:       63,
 		SinRescal:    2,
-		CtSLevel:     []uint64{13, 12},
-		StCLevel:     []uint64{3, 3},
+		ArcSineDeg:   0,
 		MaxN1N2Ratio: 16.0,
+		BitReversed:  false,
 	},
 }
+
+func computeRoots(N int) (roots []complex128) {
```

Comment:
These aren't general purpose complex methods, they are only used to generate the CtS and StC encoding matrices. We could however create a separate file for this, since you also proposed to have those two methods be part of the evaluator API (which wouldn't make them specific to the bootstrapping anymore).

Reply:
Ok so let's do that at a later stage.